### ISSUE TITLE
[BUG] `ClassConstructor` is reused across `[MethodDataSource]` tests

### ISSUE THREAD

**Issue by viceroypenguin** [thomhurst/TUnit#1007]
Consider the following scenario
```cs
using Microsoft.Extensions.DependencyInjection;
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Threading.Tasks;
using TUnit.Core.Interfaces;

public sealed class DependencyInjectionClassConstructor : IClassConstructor, ITestEndEvent
{
    private static readonly IServiceProvider services = BuildServiceProvider();
    private AsyncServiceScope asyncServiceScope;

    public T Create<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] T>() where T : class
    {
        asyncServiceScope = services.CreateAsyncScope();
        return ActivatorUtilities.GetServiceOrCreateInstance<T>(asyncServiceScope.ServiceProvider);
    }

    public async ValueTask OnTestEnd(TestContext testContext)
        => await asyncServiceScope.DisposeAsync();

    private static ServiceProvider BuildServiceProvider()
    {
        var services = new ServiceCollection();
        services.AddScoped<Dependency>();

        return services.BuildServiceProvider();
    }
}

public sealed class Dependency;


[ClassConstructor<DependencyInjectionClassConstructor>]
[NotInParallel]
public sealed class Dummy(IServiceProvider services)
{
    [Test]
    [MethodDataSource(nameof(GetValues))]
    public async Task DoSomething(int value)
    {
        await Task.Delay(1);
        var x = ActivatorUtilities.GetServiceOrCreateInstance<Dependency>(services);
    }

    public IEnumerable<int> GetValues() => [1, 2, 3, 4];
}
```

This will fail with an object disposed exception on the final test, where the scope has now been disposed 4 times.

The generated code is as follows (trimmed):
```cs
try
{
  var classArg = new DependencyInjectionClassConstructor();
  foreach (var methodData in resettableClassFactory.Value.GetValues())
  {
    ...
  }
}
```

The problem is that now the intended per-test instance data in `DependencyInjectionClassConstructor` is shared across multiple method-data tests.

**Comment by thomhurst** [comment]
Should be resolved in `0.2.26`

**Comment by viceroypenguin** [comment]
Thank you!
